mlos_bench/mlos_bench/tests/services/remote/ssh/test_ssh_host_service.py

```diff
@@ -0,0 +1,207 @@
+from mlos_bench.tests.services.remote.ssh import (SshTestServerInfo,
+                                                  ALT_TEST_SERVER_NAME,
+                                                  SSH_TEST_SERVER_NAME,
+                                                  wait_docker_service_socket)
+
+
+@requires_docker
+@pytest.mark.xdist_group("ssh_test_server")
+def test_ssh_service_remote_exec(ssh_test_server: SshTestServerInfo,
+                                 alt_test_server: SshTestServerInfo,
+                                 ssh_host_service: SshHostService) -> None:
+    """
+    Test the SshHostService remote_exec.
+
+    This checks state of the service across multiple invocations and states to
+    check for internal cache handling logic as well.
+    """
+    # pylint: disable=protected-access
+
+    config = ssh_test_server.to_ssh_service_config()
+
+    # Ensure that the SSH client isn't cached yet. Note: this may interact with
+    # state from other tests, so we group them into the same xdist_group, run
+    # with --dist loadgroup, and gc.collect() or other forms of cleanup between
+    # certain operations.
+    ssh_host_service.clear_client_cache()
+    connection_id = SshClient.id_from_params(ssh_test_server.to_connect_params())
+    assert ssh_host_service._event_loop_thread_ssh_client_cache is not None
+    connection_client = ssh_host_service._event_loop_thread_ssh_client_cache._cache.get(connection_id)
+    assert connection_client is None
+
+    (status, results_info) = ssh_host_service.remote_exec(
+        script=["hostname"],
+        config=config,
+        env_params={},
+    )
+    assert status == Status.PENDING
+    assert "asyncRemoteExecResultsFuture" in results_info
+    status, results = ssh_host_service.get_remote_exec_results(results_info)
+    assert status == Status.SUCCEEDED
+    assert results["stdout"].strip() == SSH_TEST_SERVER_NAME
+
+    # Check that the client caching is behaving as expected.
+    connection, client = ssh_host_service._event_loop_thread_ssh_client_cache._cache[connection_id]
+    assert connection is not None
+    assert connection._username == ssh_test_server.username
+    assert connection._host == ssh_test_server.hostname
+    assert connection._port == ssh_test_server.get_port()
+    local_port = connection._local_port
+    assert local_port
+    assert client is not None
+    assert client._conn_event.is_set()
+
+    # Connect to a different server.
+    (status, results_info) = ssh_host_service.remote_exec(
+        script=["hostname"],
+        config=alt_test_server.to_ssh_service_config(),
+        env_params={
+            "UNUSED": "unused",  # unused, making sure it doesn't carry over with cached connections
+        },
+    )
+    assert status == Status.PENDING
+    assert "asyncRemoteExecResultsFuture" in results_info
+    status, results = ssh_host_service.get_remote_exec_results(results_info)
+    assert status == Status.SUCCEEDED
```

Comment:
```suggestion
    assert status.is_succeeded()
```
if we change all checks like this, we can get by without
```python
from mlos_bench.environments.status import Status
```

Reply:
Work for me, though that doesn't autocomplete for some reason in the editor :shrug: